### ISSUE TITLE
Quest requirement correct but cannot load quest correctly 

### ISSUE THREAD

**Issue by hkkongou** [PikaMug/Quests#1530]
**What is your Quests version / build number (do _not_ say "latest")?**
Quests-4.0.0-rc.3.jar 1.8.6
--

**Describe the problem and how to replicate it. Any console errors to post _using Pastebin.com_?**

upgrade Quests-4.0.0-rc.3.jar from 178 to 179, it show error ``Requirement quests has unknown quest name NULL, see quest of ID custom1103 `` i know this is a bug fix from 179 , however it seems like lead to the following problem. 
To test this problem, i removed quests down to only 3 quest.

Quest requirement maybe broke if the quest name is using foreign language, I use Chinese as quest name. I set up a quest with requirement of another quest, but it say the requirement quest quest name is NULL, which does not make sense as the quest is correctly loading, if I remove the quest requirement,  it load normally.

I saw the update of #179 changed will Throw error if requirement quest name is unknown, but the requirement name is correct. 

entire server log 
https://pastebin.com/AbyC3GLs

bug part
https://pastebin.com/C4R0dCzU

originally, i have 300+ quest but use to demonstrate the bug, i cut the quests until only have 3 in my dev server. 

![image](https://user-images.githubusercontent.com/61608872/103397157-00666e80-4b72-11eb-8862-cb055b271aa0.png)

--

**If applicable, can you provide an example quest from quests.yml _using Pastebin.com_?**

quest.yml:

https://pastebin.com/uLgF01WX

**Comment by DeadlykillOfficial** [comment]
You have manually edited your quests.yml file therefor it drops errors.

**Comment by hkkongou** [comment]
> You have manually edited your quests.yml file therefor it drops errors.

yes, but actually no, there is no way for me to remove 300 quest become 3 to test this bug

**Comment by DeadlykillOfficial** [comment]
First of all you should always keep up to date your plugin. Build 186 is current latest one, relating to all bugfixes.
Secondly the issue is caused because you most likely self edited ItemStacks within quests.yml
When using editor ItemStacks never generate `&` see #1527

**Comment by DeadlykillOfficial** [comment]
It's not a syntax error in this case. It's unsupported characters being used within it.
This is why you should never ever edit your quests.yml file unless you want to fix a typo or any text somewhere.

**Comment by hkkongou** [comment]
> First of all you should always keep up to date your plugin. Build 186 is current latest one, relating to all bugfixes.
> Secondly the issue is caused because you most likely self edited ItemStacks within quests.yml
> When using editor ItemStacks never generate `&` see #1527

1. i forgot to say, i have tested from 186 to 170 , each version, this problem start from #179 to #186 build
2. i am already using quest editor to set up requirement

**Comment by DeadlykillOfficial** [comment]
The way Spigot works is pretty basic when using names within YML there are few basic rules.
Your Item name cannot start with special character also it must be within quotes.
try to use `ยง5` as color tag on the name and make sure to `' '` quote it.
So it becomes like `display-name: 'ยง1 LV 1  頭盔'`

**Comment by hkkongou** [comment]
> The way Spigot works is pretty basic when using names within YML there are few basic rules.
> Your Item name cannot start with special character also it must be within quotes.
> try to use `ยง5` as color tag on the name and make sure to `' '` quote it.
> So it becomes like `display-name: 'ยง1 LV 1 頭盔'`

These items are generated by mythicmobs and is not manually edited by anyone using external method, we used the quest editor to load the item and save it properly, if you say the color tag will become 'ยง1 LV 1 頭盔' , then why not the quest system done it automatically for me? you are telling me to edit the file by hand now but last comment you tell me to use in game editor.

**Comment by DeadlykillOfficial** [comment]
I'm not saying to edit it everytime. This right now is literally just investigating where the exact issue is. This is why I asked you to edit it. So you can test it out and see. Normally quests should auto set up the character but from other hand I'm using b186 myself. So I can't tell it 179 or any older would normally generate proper sign. As everyday there are different fixes. I'm trying to help you investigate if its color issue or Chinese characters issue. It leads me to color code issue because normally those signs are unsupported and cause issues in quests. Also it say quest name is null. Perhaps check on that name in requirements: quests: Chinese name. And remove it and test it out if it works without it but the Itemstack with color codes. If that works then you either have to wait for pika to find solution for Chinese or rename your quests to English

**Comment by DeadlykillOfficial** [comment]
![Screenshot_20201231-075103_Opera beta](https://user-images.githubusercontent.com/12415714/103398427-fd04c000-4b3c-11eb-8e18-98e4f2be5502.jpg)
This might be the issue specifically not the Itemstacks at all. Check on this and test it out using English related names only. If it works then it's Chinese characters issues and you gotta wait on pika for a fix

**Comment by hkkongou** [comment]
ok, i try to change it to english, thank you for your suggestions, i am sorry that i goes emotional

**Comment by hkkongou** [comment]
i start to see why this problem happen, we have been digging a wrong way, english or chinese language is not related.

I suspicious the quest report NULL is due to the secondary quest is before the primary quest.
![image](https://user-images.githubusercontent.com/61608872/103398673-b9c84280-4b78-11eb-911f-349b0f2ebd26.png)

The above picture show 3 quest.
custom 1101 is primary, linked with 1103 and 1102 with quest requirement 1101.
as custom1103 load, 1101 haven't load and 1103 cannot find 1101.
So it say 1103 quest is null.

If so, quest 179 version maybe changed how the quests load, before can be randomly loaded, but after 179, all quest are loaded in sequence.

**Comment by PikaMug** [comment]
Thanks for the report. The chosen language should not make any difference. I would think the order of quests in the file should not matter either.

Since you have (correctly) updated to the latest development build, could you post the new console error? The line numbers will be different since it's a newer version.

**Comment by hkkongou** [comment]
yes i think the order of the quest is the problem.

i have moved the custom1103 entirely before custom1102, now it loaded.
However, this can only be done with text editor but not in game.

This would be a big problem if quests are designed modulely in teams, if a quest is designed then add a quest requirement which created after, the quest cannot load.

**Comment by hkkongou** [comment]
plz wait, i try to load with #186

**Comment by hkkongou** [comment]
same in #186
order of quests:
![image](https://user-images.githubusercontent.com/61608872/103399059-3dcefa00-4b7a-11eb-9657-8582ff13ac32.png)

![image](https://user-images.githubusercontent.com/61608872/103399051-37408280-4b7a-11eb-9d54-e13b2f5ca318.png)

**Comment by hkkongou** [comment]
however, no problem with loading if order is
![image](https://user-images.githubusercontent.com/61608872/103399080-53dcba80-4b7a-11eb-803c-d108ede50a06.png)


![image](https://user-images.githubusercontent.com/61608872/103399075-4fb09d00-4b7a-11eb-8a71-0034fcd0fd7f.png)

**Comment by PikaMug** [comment]
I appreciate you being so thorough. Yes, it would seem the order does matter for those with a Requirement quest or quest blocks. Not sure if there's an easy way to fix this, but I'll look into it.

**Comment by hkkongou** [comment]
i am sorry, i have read your fix which will show this message if this error occurs:
![image](https://user-images.githubusercontent.com/61608872/103399825-d6b34480-4b7d-11eb-8c46-6f88ee03a5ca.png)
however, in my server i use quest build in item gui to display the quest in order, to make quest in item set up in the following order, i have to more some quest after another quest, which will lead to the above bug
![image](https://user-images.githubusercontent.com/61608872/103399849-f185b900-4b7d-11eb-8f8c-9b139b6b81ae.png)
The above highlighed quest will unlock all other quest in this gui requirement.

**Comment by hkkongou** [comment]
This gui order in gui is design to have special meaning, if the required quest is moved to front, the meaning will lost
![image](https://user-images.githubusercontent.com/61608872/103399924-3c9fcc00-4b7e-11eb-86ed-902835e14448.png)

**Comment by hkkongou** [comment]
should i open another issue bug or suggestion again 🤔

**Comment by PikaMug** [comment]
I opted for a more detailed error message because I couldn't see a way to properly address the issue. Even if a proper fix were presented, there's no guarantee that the load order will remain the same going forward. GUI ordering is really something that should be addressed with QuestsGUI eventually. You [posted about this in the discussion thread a while ago](https://www.spigotmc.org/threads/questsgui-paid.399346/page-7#post-4023135); I have not forgotten.

**Comment by hkkongou** [comment]
> I opted for a more detailed error message because I couldn't see a way to properly address the issue. Even if a proper fix were presented, there's no guarantee that the load order will remain the same going forward. GUI ordering is really something that should be addressed with QuestsGUI eventually. You [posted about this in the discussion thread a while ago](https://www.spigotmc.org/threads/questsgui-paid.399346/page-7#post-4023135); I have not forgotten.

does that mean you are leaving me 2 options...
1. fixed the order in order to update latest version, but gui also need to change 
2. stay # 178, do not upgrade any further

**Comment by hkkongou** [comment]
This problem does not occur in # 178 or other version of quest, i saw # 179 build change due to some bugs(?) or not? will it be possible for using another method to fix the bug so that the next patch, the requirement check also any order 🤔🤔
